### DISCUSSION TITLE
What are differences between rdDistGeom.EmbedMolecule and AllChem.EmbedMolecule?

### DISCUSSION THREAD

**Discussion by mainguyenanhvu** [rdkit/rdkit#7923] [Q&A] [answered]
I am trying to use RDKit to convert SMILES into 3D SDF file. I read several posts to solve `Bad conformer ID` when using AllChem.UFFOptimizeMolecule.

Then, I saw [a post](https://greglandrum.github.io/rdkit-blog/posts/2023-05-17-understanding-confgen-errors.html) used rdDistGeom.EmbedMolecule to generate conformations. I would like to know how and when to use them.

**Comment by greglandrum** [comment, marked as answer]
The two functions are exactly the same thing.
There's a bunch of information out there for how to use the RDKit conformer generation code. If you have read through that and have specific questions, feel free to ask them.
